### ISSUE TITLE
Switching resources in Edit/Create uses fields from original resource

### ISSUE THREAD

**Issue by braco** [marmelab/react-admin#7017]
We're using ra-supabase.

For the `<List />` we have a Postgres view (original_table + join to another table for extra columns).

For Edit and Create, we need to switch back to original_table so it can be updated directly.

```
<Edit resource="original_table" />
```

Switching resources like this throws an error because it's trying to select columns from the view, it doesn't switch to the columns in original_table.

```
<ResourceContextProvider value="original_table">
    <Edit />
</ResourceContextProvider>
```
Wrapping in ResourceContextProvider doesn't seem to switch the resource at all.

Is there any pattern that works?

**Comment by fzaninotto** [comment]
Hi, and thanks for your report. 

> Switching resources like this throws an error because it's trying to select columns from the view

Can you be more specific, and link to a CodeSandbox demonstrating the issue?

**Comment by braco** [comment]
Hi @fzaninotto I'm not sure how to make a codesandbox for this since it would necessitate open access to a sql database. Maybe I'll see if I can recreate it with mocks.

I'm trying to work around some of the limitations of React Admin. It's easier to make Lists based on [SQL views](https://www.postgresql.org/docs/13/tutorial-views.html), which gives us good ergonomics for filtering/sorting with complex relationships to other tables. Then all the complexity lives on the SQL server and doesn't need to be reflected in the front end.

However, when the user goes to edit a record, it should be editing original table directly, not the SQL view. Setting up logic to edit views is painful on most SQL servers. So we just need to switch the user back to the original table somehow.

Given this setup to `<List />` the SQL view:
```
  <Admin>
    <Resource name="sql_view" list={List} edit={Edit} />
    <Resource name="original_table" edit={Edit} ... />
  </Admin>
```

We want the `edit` action in `sql_view` to switch to `original_table`.

I tried this:

```
<Edit resource="original_table" />
```

^ when entering into the edit screen, there's fatal error: React Admin is trying to select columns from `sql_view` that don't exist in `original_table`. It's switching the resource to `original_table` but still requesting fields from `sql_view`.

and this:
```
const Edit = () => 
    <ResourceContextProvider value="original_table">
         ....
    </ResourceContextProvider>
```

Which doesn't seem to work at all, the user is still editing `sql_view`.

I also tried the opposite, using `original_table` and trying to switch the View action to `sql_view`.
